python/lsst/obs/base/ingest.py

```diff
@@ -479,6 +530,39 @@ def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Op
             if this_run not in runs:
                 self.butler.registry.registerCollection(this_run, type=CollectionType.RUN)
                 runs.add(this_run)
-            with self.butler.transaction():
-                refs.extend(self.ingestExposureDatasets(exposure, run=this_run))
+            try:
+                with self.butler.transaction():
+                    refs.extend(self.ingestExposureDatasets(exposure, run=this_run))
+            except Exception as e:
+                n_ingests_failed += 1
+                self.log.warning("Failed to ingest the following files for reason: %s", e)
+                for f in exposure.files:
+                    self.log.warning("- %s", f.filename)
+                continue
+
+            # Success for this exposure
+            n_exposures += 1
+            self.log.warning("Exposure %s:%s ingested successfully",
+                             exposure.record.instrument, exposure.record.name)
+
+        had_failure = False
+        self.log.info("Processed data from %d exposure%s with %d failure%s from exposure registration"
+                      " and %d failure%s from file ingest.",
+                      n_exposures, "" if n_exposures == 1 else "s",
+                      n_exposures_failed, "" if n_exposures_failed == 1 else "s",
+                      n_ingests_failed, "" if n_ingests_failed == 1 else "s")
+        if n_exposures_failed > 0 or n_ingests_failed > 0:
+            had_failure = True
+        self.log.info("Ingested %d distinct Butler dataset%s",
+                      len(refs), "" if len(refs) == 1 else "s")
+
+        if self._failed_metadata_files:
+            had_failure = True
+            self.log.warning("Could not extract metadata from the following:")
+            for f in self._failed_metadata_files:
+                self.log.warning("- %s", f)
```

Comment:
Given that these are the failures we catch first, it might be better to report them before the other kinds of failures, too, even though they're all deferred.  It might even make sense to report these failures before we start any database writes, but that's a tradeoff between reporting failures early and having a good error summary in the tail, and I don't have a good sense of what matters more here.

Reply:
I can move those first. I think I do want to guarantee that these reports all turn up at the end though so that in debug logging mode you aren't swamped with other information and have to hunt for the relevant information.